File: api/src/main/java/net/countercraft/movecraft/craft/SubCraft.java
package net.countercraft.movecraft.craft;

import net.countercraft.movecraft.util.MathUtils;
import org.bukkit.block.TileState;
import org.bukkit.persistence.PersistentDataType;
import org.jetbrains.annotations.NotNull;

public interface SubCraft extends Craft {

    @NotNull
    Craft getParent();

    void setParent(@NotNull Craft parent);

    @Override
    default void removeUUIDMarkFromTile(TileState tile) {
        Craft parent = this.getParent();
        if (parent != null) {
            tile.getPersistentDataContainer().set(
                    MathUtils.KEY_CRAFT_UUID,
                    PersistentDataType.STRING,
                    parent.getUUID().toString()
            );
        } else {
            Craft.super.removeUUIDMarkFromTile(tile);
        }
    }
}

File: src/main/kotlin/dev/cccm5/movecraft/craft/SubCraft.kt
package dev.cccm5.movecraft.craft

import dev.cccm5.movecraft.hitbox.NonEmptyHitBox
import dev.cccm5.movecraft.util.Direction
import org.bukkit.World

class SubCraft(override var hitBox: NonEmptyHitBox, override var orientation: Direction, override var world: World) : Craft() {

}